### ISSUE TITLE
questions for netmoss score plot

### ISSUE THREAD

**Issue by chaichai9521** [xiaolw95/NetMoss2#8]
Hi,

I just find a strange problems. When ran same input dataset twice, I got totally different netmoss score plot. For example, at the 1st time running, bacteria1 with the highest netmoss score was listed. However, at the 2nd time running, bacteria3 with the highest netmoss score and bacteria1 disappeared. 

So, do we need run "set.seed()" command before run netmoss?

thanks
Jianmin

**Comment by xiaolw95** [comment]
Are the network matrices identical for both tests? The NetMoss score plot only exhibits the top 30 differential bacteria and you may check the NetMoss score list to see whether bacteria1 stays.

**Comment by chaichai9521** [comment]
yeah. The network matrices were identical for both test. I did export netmoss score list. I found the netmoss score for bacterial 1  is 1 at the first time running, But, at the second time running, bacteria 1 netmoss score is 0.64. So, I want to know why did this happen? As you know, bacteria 1 in my study is an important pathogen, so how do I trust netmoss algorithm?  Could you give me an better explanation or solution for this? thanks.

**Comment by xiaolw95** [comment]
Is it possible for you to send the input data to me? This situation did not occur in other datasets, and I am not sure what was going on here.

**Comment by chaichai9521** [comment]
Feature1_Mycoplasma_hyorhinis


> Is it possible for you to send the input data to me? This situation did not occur in other datasets, and I am not sure what was going on here.

Please see attached file that contain original inputs, R code and netmoss score list. Please keep my files confidential. And maybe you can add my wechat (517289573) for better communication. thanks.

[New folder.zip](https://github.com/xiaolw95/NetMoss2/files/8989655/New.folder.zip)

**Comment by xiaolw95** [comment]
Copy that! I shall keep in touch with you and you can delete the files now.

**Comment by xiaolw95** [comment]
> yeah. The network matrices were identical for both test. I did export netmoss score list. I found the netmoss score for bacterial 1 is 1 at the first time running, But, at the second time running, bacteria 1 netmoss score is 0.64. So, I want to know why did this happen? As you know, bacteria 1 in my study is an important pathogen, so how do I trust netmoss algorithm? Could you give me an better explanation or solution for this? thanks.

I still didn't get it because for my test the result is reproducible even when I use your dataset. I have to confirm with you again **if you rebuilt ecological networks each time** using the netBuild function. **NOTE: I refer to the network matrix, not the abundance matrix.**
If it's true for you then it's not weird you got a different score because the network matrix altered. The NetMoss algorithm relies heavily on the input networks. In our tutorial example, only a guide on simplified network construction was provided and you can refer to [SparCC](https://github.com/bio-developer/sparcc) or [SPIEC-EASI](https://github.com/zdk123/SpiecEasi) for more reliable approaches as we have mentioned.

**Comment by chaichai9521** [comment]
OK. I will try SparCC or SPICE-EASI. 
yeah, I built network matrix every time. So, based on your helpful comments, "The NetMoss algorithm relies heavily on the input networks." This is easily understood for me. But, if you import same dataset of bacterial abundances, why you got a different network matrix and subsequently caused different netmoss score list? Is this a bug for NetMoss algorithm? In theory, same input abundance data should get same netmoss score list, right ? Sorry to trouble you, but I believe the output from same dataset should be consistent.

**Comment by xiaolw95** [comment]
NetMoss score is used to evaluate the difference in network structure. For this reason, different network topologies lead to different results even if the relative abundance matrix does not alter. As for why it occurs and how much difference might exist between different batches, it is concerned with the network construction method you choose. Microbial network construction is challenged by multiple factors and you may refer to some papers for more information.
[1](https://www.nature.com/articles/s41396-021-01027-4)
[2](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002687)
[3](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004226)
